src/code/z_viscvg.c

```diff
@@ -1,70 +1,100 @@
-#include "global.h"
+/**
+ * @file z_viscvg.c
+ * Description: Visualise Coverage (how much of the neighbourhood of a pixel is covered by a primitive) in various ways.
+ *
+ * To understand this file, it is helpful to remember that A_MEM is essentially synonymous with coverage, and that
+ * `GBL_c1/2(p, a, m, b)` are usually `(p * a + m * b) / (a + b)`.
+ *
+ * Coverage is maximum when not on an edge, on an edge it is smaller, and since coverage is treated as an alpha edges
+ * show up as darker in all of the available modes.
+ *
+ * We abbreviate coverage to "cvg"; "pixel RGB" is the original colour the pixel had before the filter is applied.
+ */
 
-// Note : This file is related to z_vismono, the original name was probably z_vis<something before "mono"
-// alphabetically>
+#include "global.h"
 
-Gfx D_8012AC00[] = {
+/**
+ * Draws only coverage: does not retain any of the original pixel RGB, primColor is used as background colour.
+ */
+Gfx sCoverageOnlyDL[] = {
     gsDPSetOtherMode(G_AD_PATTERN | G_CD_MAGICSQ | G_CK_NONE | G_TC_CONV | G_TF_POINT | G_TT_NONE | G_TL_TILE |
                          G_TD_CLAMP | G_TP_NONE | G_CYC_1CYCLE | G_PM_NPRIMITIVE,
                      G_AC_NONE | G_ZS_PRIM | G_RM_VISCVG | G_RM_VISCVG2),
+    // (blendColor) * (cvg)
     gsDPFillRectangle(0, 0, SCREEN_WIDTH - 1, SCREEN_HEIGHT - 1),
     gsDPPipeSync(),
     gsDPSetBlendColor(0, 0, 0, 8),
     gsSPEndDisplayList(),
 };
 
-Gfx D_8012AC28[] = {
+/**
+ * Draws fog + coverage * RGB of pixels
+ *
+ * @bug This easily overflows the blender because the fog value is added to the coverage value.
+ */
+Gfx sCoverageRGBFogDL[] = {
     gsDPSetOtherMode(G_AD_PATTERN | G_CD_MAGICSQ | G_CK_NONE | G_TC_CONV | G_TF_POINT | G_TT_NONE | G_TL_TILE |
                          G_TD_CLAMP | G_TP_NONE | G_CYC_1CYCLE | G_PM_NPRIMITIVE,
                      G_AC_NONE | G_ZS_PRIM | IM_RD | CVG_DST_CLAMP | ZMODE_OPA | FORCE_BL |
                          GBL_c1(G_BL_CLR_FOG, G_BL_A_FOG, G_BL_CLR_MEM, G_BL_A_MEM) |
                          GBL_c2(G_BL_CLR_FOG, G_BL_A_FOG, G_BL_CLR_MEM, G_BL_A_MEM)),
+    // (fog RGB) * (fog alpha) + (pixel RGB) * (cvg)
     gsDPFillRectangle(0, 0, SCREEN_WIDTH - 1, SCREEN_HEIGHT - 1),
     gsSPEndDisplayList(),
 };
 
-Gfx D_8012AC40[] = {
+/**
+ * Draws coverage and RGB of pixels
+ */
+Gfx sCoverageRGBDL[] = {
     gsDPSetOtherMode(G_AD_PATTERN | G_CD_MAGICSQ | G_CK_NONE | G_TC_CONV | G_TF_POINT | G_TT_NONE | G_TL_TILE |
                          G_TD_CLAMP | G_TP_NONE | G_CYC_1CYCLE | G_PM_NPRIMITIVE,
                      G_AC_NONE | G_ZS_PRIM | IM_RD | CVG_DST_CLAMP | ZMODE_OPA | FORCE_BL |
                          GBL_c1(G_BL_CLR_IN, G_BL_0, G_BL_CLR_MEM, G_BL_A_MEM) |
                          GBL_c2(G_BL_CLR_IN, G_BL_0, G_BL_CLR_MEM, G_BL_A_MEM)),
```

Comment:
Can't this and the one below be replaced with G_RM_VISCVG ?

Reply:
No, because
```
#define RM_VISCVG(clk) \
    IM_RD | FORCE_BL |     \
    GBL_c##clk(G_BL_CLR_IN, G_BL_0, G_BL_CLR_BL, G_BL_A_MEM)
```
uses blendColor, not mem.